cmd/snap/cmd_auto_import.go

```diff
@@ -57,6 +57,32 @@ func autoImportCandidates() ([]string, error) {
 		if len(l) == 0 {
 			continue
 		}
+		if len(l) < 7 {
+			return nil, fmt.Errorf("cannot parse line %q: too short", scanner.Text())
+		}
+
+		// see proc.txt:3.5 /proc/<pid>/mountinfo - the field (7)
+		// can have variable length, so we need to find the "-"
+		// separator
+		var i int
+		for i = 6; i < len(l)-1 && l[i] != "-"; i++ {
+			// nothing
+		}
+		if i == len(l)-1 {
+			return nil, fmt.Errorf("cannot parse line %q: no separator '-' found", scanner.Text())
+		}
+
+		mountSrc := l[i+2]
```

Comment:
This is not safe given the tests above. If `len(l) == 10 && i == 8`, then `i < len(l)-1`, yet `l[i+2]` is a panic.